### ISSUE TITLE
[Feature Request] Adding references to example input scripts in the doc pages

### ISSUE THREAD

**Issue by ndtrung81** [lammps/lammps#2680]
**Summary**

It would be helpful and instructive if the doc page of each command (fixes, pair styles, commands) contains pointers (or references) to some, if not all, input scripts in the examples/ sub-folders that illustrate the use of the command.

**Detailed Description**

In many cases, users would search for a command (on the web or in the documentation) and try to find an example input script for it. With some pointers (or references) to the corresponding input scripts (i.e. under which sub-folder in examples) in the doc page, users can find what they need faster. Also, when some developers contribute new codes and some example scripts, they can point users to these new examples, too.

For example, in the doc page of fix colvars, please include a section: 

Example input scripts available: examples/USER/colvars

or in doc page of fix ave/correlate 

Example input scripts available: examples/VISCOSITY; examples/KAPPA

**Further Information, Files, and Links**

The current workaround is to use the grep command in the examples folder, but we can do better than this, right?

**Comment by agiliopadua** [comment]
Of course. I can do that for USER-FEP and USER-DRUDE.

**Comment by sjplimp** [comment]
yes, this is great idea.  I know we've made this reference on a handful of doc pages, but nothing systematic.  Given the suggested syntax, I think it would best appear at the top of each page, maybe after "Examples" of the command syntax itself.

**Comment by sjplimp** [comment]
Here is what we have currently:  cd lammps/doc/rst; grep "examples/" [a-z]*.rst
Your one line syntax is better.

**Comment by sjplimp** [comment]
@rbberger Do you want to write a Python script to scan the example dirs for style commands, and auto-add these one-liners to doc/*.rst files?

**Comment by akohlmey** [comment]
> yes, this is great idea. I know we've made this reference on a handful of doc pages, but nothing systematic. Given the suggested syntax, I think it would best appear at the top of each page, maybe after "Examples" of the command syntax itself.

I second that idea. This is a perfect task for volunteers that want to contribute to LAMMPS

**Comment by akohlmey** [comment]
> @rbberger Do you want to write a Python script to scan the example dirs for style commands, and auto-add these one-liners to doc/*.rst files?

Not so sure about this idea. Just because some example uses a LAMMPS command doesn't make it automatically a good example for that command. There should only be references added where the example provides significantly more information than the example command lines we already have directly in the .rst file.

**Comment by sjplimp** [comment]
@rbberger @akohlmey So what about a python script where you specify an rst filename and it prints a list of example dir input scripts that use any of the commands in that rst file.  With an option to auto-insert that info in the rst file (or maybe a sub-selection thereof).

**Comment by rbberger** [comment]
We could make them link to the folder/files on GitHub. E.g.

> Example input scripts available: [examples/VISCOSITY](https://github.com/lammps/lammps/tree/master/examples/VISCOSITY), [examples/KAPPA](https://github.com/lammps/lammps/tree/master/examples/KAPPA)

That way we don't have to have yet-another-hosting of the examples folder in some XYZ version to keep in sync.

**Comment by akohlmey** [comment]
> We could make them link to the folder/files on GitHub. E.g.
> 
> > Example input scripts available: [examples/VISCOSITY](https://github.com/lammps/lammps/tree/master/examples/VISCOSITY), [examples/KAPPA](https://github.com/lammps/lammps/tree/master/examples/KAPPA)
> 
> That way we don't have to have yet-another-hosting of the examples folder in some XYZ version to keep in sync.

I think it would be sufficient to just mention the folder(s). Having a working URL would be quite complex, since it needs to work not just for online versions of the docs (and there are not just the ones we host on the web) which will have to point to specific (released) versions of the source, but also the local version that is packages with the .tar.gz files from lammps.sandia.gov and that people may view locally.

**Comment by sjplimp** [comment]
I'll just add that it should not point to folders, it should point to specific input scripts.  Some folders have multiple scripts.

**Comment by jtclemm** [comment]
I'd be willing to work on this if there's still a need/interest. I've often wished there was an easier way to identify relevant examples. Condensing the above discussion, maybe a format like this https://github.com/jtclemm/lammps/blob/documentation-examples/doc/src/angle_charmm.rst ?

**Comment by sjplimp** [comment]
@jtclemm I like that format
One tweak:
Used in input scripts:
instead of
Example input scripts avail:

**Comment by sjplimp** [comment]
@akohlmey @jtclemm
Also were those input script names going to be links?  Is that easy/possible?

**Comment by jtclemm** [comment]
@sjplimp I updated the example and added links.

I'll go ahead and write a python script to generate a list of all possible input scripts and start manually filtering out unnecessary examples (e.g. for fix nve). When the format/URLs are agreed upon I'll edit the doc files.

**Comment by akohlmey** [comment]
> @akohlmey @jtclemm
> Also were those input script names going to be links? Is that easy/possible?

I don't see a clean way to link those inputs. The GitHub master branch is the only consistent URL but that may be different from the version of the docs you are looking at and thus the file may have been moved or renamed or not longer exist.

**Comment by akohlmey** [comment]
> @sjplimp I updated the example and added links.
> 
> I'll go ahead and write a python script to generate a list of all possible input scripts and start manually filtering out unnecessary examples (e.g. for fix nve). When the format/URLs are agreed upon I'll edit the doc files.

It would be better to wait a little bit. We just started some reorganization of packages and correspondingly also the example folders. Whatever you would do now will likely have significant merge conflicts. Better 2-3 weeks when we have a first release candidate version and switch to bugfix-only code changes until the next stable release is out.

**Comment by akohlmey** [comment]
@jtclemm please note that we just today released a new patch release which completes the reorganization of packages in LAMMPS and thus also examples should be now in a more-or-less final location (individual folders can be changed if it makes sense).

**Comment by jtclemm** [comment]
@akohlmey Thanks for the reminder, I'll go back and look at the docs.

**Comment by jtclemm** [comment]
I have an updated set of doc files in [this branch](https://github.com/jtclemm/lammps/tree/documentation-examples)  . For  all commands that were present in an example script (excluding some trivial commands like run, read_restart, etc) there is now a section in the documentation that lists some of those scripts. For instance [compute temp/region](https://github.com/jtclemm/lammps/blob/155593dbb44225ece83544bd603caa02271920f4/doc/src/compute_temp_region.rst) now lists examples/HEAT/in.lj.ehex and examples/KAPPA/in.ehex.

Some commands were present in a lot of input scripts (such as lj cut), in those cases only a few examples were pseudo-arbitrarily included in the documentation. I tried to prioritize input scripts that weren't in the PACKAGES folder and a few simple scripts I was familiar with (in.melt), but the process was otherwise mostly random/automated. Are there any other ideas on how to choose which examples to include such as favoring shorter scripts? Or maybe just listing some random scripts is good enough?
